### DISCUSSION TITLE
Friday night funkin kade engine 1.6 how to change the font language

### DISCUSSION THREAD

**Discussion by Reyororo** [KadeArchive/Kade-Engine#1595] [Q&A] [answered]
I want the dialogue font to be Japanese. How can I make it into Japanese?

**Comment by isakube** [comment]
good question. unfortunately, no.

**Reply by Reyororo** [reply]
> 良い質問。残念だけど違う。

I want to change the language like this mod
https://gamebanana.com/mods/287852

**Comment by napstaa967** [comment]
> I want the dialogue font to be Japanese. How can I make it into Japanese?

you can't

the default fnf font only has A-Z and some symbols

you have to draw the japanese characters yourself and code them in

**Reply by Reyororo** [reply]
Rather than changing the font of theintrotext. I want to change the text of the dialogue.
Like this mod ↓
https://gamebanana.com/mods/287852

**Reply by Riskerman21** [reply]
you can edit the .txt files inside assets/preload/data/thorns/thorns-dialouge.txt

**Reply by Reyororo** [reply]
> you can edit the .txt files inside assets/preload/data/thorns/thorns-dialouge.txt

I'm sorry I made a mistake I want to change the text font of the dialogue to Japanese

**Reply by Riskerman21** [reply]
bro you have to draw them
stoop complaining

![alphabet](https://user-images.githubusercontent.com/61993263/128596791-fbd43d04-2603-4b60-b7d3-b23376ce46e0.png)

**Reply by Reyororo** [reply]
> bro you have to draw them
> stoop complaining
> 
> ![alphabet](https://user-images.githubusercontent.com/61993263/128596791-fbd43d04-2603-4b60-b7d3-b23376ce46e0.png)

I don't want to change the font of the intro text I want to change the text font of the dialog

**Reply by Riskerman21** [reply]
here in dialoguebox.hx

find the code:
		


            dropText = new FlxText(242, 502, Std.int(FlxG.width * 0.6), "", 32);
		dropText.font = 'Pixel Arial 11 Bold';
		dropText.color = 0xFFD89494;
		add(dropText);

		swagDialogue = new FlxTypeText(240, 500, Std.int(FlxG.width * 0.6), "", 32);
		swagDialogue.font = 'Pixel Arial 11 Bold';
		swagDialogue.color = 0xFF3F2021;
		swagDialogue.sounds = [FlxG.sound.load(Paths.sound('pixelText'), 0.6)];
		add(swagDialogue);


okei, now edit the part that says,
Pixel Arial 11 Bold

into whatever font you want

pls add this as the answer if it helped.

**Reply by Reyororo** [reply]
> here in dialoguebox.hx
> 
> find the code:
> 
> ```
>         dropText = new FlxText(242, 502, Std.int(FlxG.width * 0.6), "", 32);
> 	dropText.font = 'Pixel Arial 11 Bold';
> 	dropText.color = 0xFFD89494;
> 	add(dropText);
> 
> 	swagDialogue = new FlxTypeText(240, 500, Std.int(FlxG.width * 0.6), "", 32);
> 	swagDialogue.font = 'Pixel Arial 11 Bold';
> 	swagDialogue.color = 0xFF3F2021;
> 	swagDialogue.sounds = [FlxG.sound.load(Paths.sound('pixelText'), 0.6)];
> 	add(swagDialogue);
> ```
> 
> okei, now edit the part that says,
> Pixel Arial 11 Bold
> 
> into whatever font you want
> 
> pls add this as the answer if it helped.

When I changed the font type to jfdot14 and made the dialogue
![スクリーンショット (4)](https://user-images.githubusercontent.com/88378158/128615052-592e2b71-f421-4009-82a6-9b1b88eb7972.png)
![スクリーンショット (6)](https://user-images.githubusercontent.com/88378158/128615054-69702ced-f6ac-43bb-9b7b-2e5fc2d151dc.png)
![スクリーンショット (7)](https://user-images.githubusercontent.com/88378158/128615056-74b8b1b0-a857-4d52-9c43-2d2c4481bb12.png)
![スクリーンショット (8)](https://user-images.githubusercontent.com/88378158/128615057-89b8822f-508b-4d7e-90dc-79c90d3f45fa.png)
 Japanese and tested it, Senpai stopped speaking.

**Reply by napstaa967** [reply]
click the font you're trying to add

read the name it has on the windows install prompt

set that as the name of the font in the code

**Reply by Reyororo** [reply]
> 追加しようとしているフォントをクリックします
> 
> Windowsのインストールプロンプトで名前を読んでください
> 
> コード内のフォントの名前としてそれを設定します

I tried it, but after all Senpai didn't talk

**Reply by Riskerman21** [reply]
Make sure you have that font downloaded and compatible with haxe bro otherwise it doesn't work

**Reply by Reyororo** [reply]
> そのフォントがダウンロードされ、haxe broと互換性があることを確認してください。そうでない場合は、機能しません。

I think it's compatible because it uses the same font type as this mod

https://gamebanana.com/mods/287852

**Reply by Riskerman21** [reply]
the input you giving it doesn't make sense to engine. Did the mod use the original fnf or Kade engine?

**Reply by Riskerman21** [reply]
Here this guy is smart:



	dropText = new FlxText(242, 502, Std.int(FlxG.width * 0.6), "", 32);
		dropText.setFormat("assets/fonts/jfdot14.ttf", dropText.size);
		dropText.color = 0xFFD89494;
		add(dropText);

		swagDialogue = new FlxTypeText(240, 500, Std.int(FlxG.width * 0.6), "", 32);
		swagDialogue.setFormat("assets/fonts/jfdot14.ttf", swagDialogue.size);
		swagDialogue.color = 0xFF3F2021;
		swagDialogue.sounds = [FlxG.sound.load('assets/sounds/pixelText' + TitleState.soundExt, 0.6)];
		add(swagDialogue);

		dialogue = new Alphabet(0, 80, "", false, true);
		// dialogue.x = 90;
		// add(dialogue);

		this.dialogueList = dialogueList;

then in the text it doesn't change:

:dad:Ah, a new fair maiden has come in search of true love!
:dad:A serenade between gentlemen shall decide where her beautiful heart shall reside.
:bf:Beep bo bop


I would clone his code here:
[https://github.com/doggy-dentures/Funkin](https://github.com/doggy-dentures/Funkin)

**Reply by Reyororo** [reply]
> Here this guy is smart:
> 
> ```
> dropText = new FlxText(242, 502, Std.int(FlxG.width * 0.6), "", 32);
> 	dropText.setFormat("assets/fonts/jfdot14.ttf", dropText.size);
> 	dropText.color = 0xFFD89494;
> 	add(dropText);
> 
> 	swagDialogue = new FlxTypeText(240, 500, Std.int(FlxG.width * 0.6), "", 32);
> 	swagDialogue.setFormat("assets/fonts/jfdot14.ttf", swagDialogue.size);
> 	swagDialogue.color = 0xFF3F2021;
> 	swagDialogue.sounds = [FlxG.sound.load('assets/sounds/pixelText' + TitleState.soundExt, 0.6)];
> 	add(swagDialogue);
> 
> 	dialogue = new Alphabet(0, 80, "", false, true);
> 	// dialogue.x = 90;
> 	// add(dialogue);
> 
> 	this.dialogueList = dialogueList;
> ```
> 
> then in the text it doesn't change:
> 
> :dad:Ah, a new fair maiden has come in search of true love!
> :dad:A serenade between gentlemen shall decide where her beautiful heart shall reside.
> :bf:Beep bo bop
> 
> I would clone his code here:
> https://github.com/doggy-dentures/Funkin

Where can I see your clone

**Reply by Reyororo** [reply]
> the input you giving it doesn't make sense to engine. Did the mod use the original fnf or Kade engine?

indeed

**Reply by Reyororo** [reply]
> Make sure you have that font downloaded and compatible with haxe bro otherwise it doesn't work

How can I make sure it is compatible with heax bro?

**Comment by Riskerman21** [comment]
Just clone the mod:[https://gamebanana.com/mods/287852](https://gamebanana.com/mods/287852)'s code:
I would clone his repository here:
[https://github.com/doggy-dentures/Funkin](https://github.com/doggy-dentures/Funkin)

good luck

**Reply by Reyororo** [reply]
> Just clone the mod:https://gamebanana.com/mods/287852's code:
> I would clone his repository here:
> https://github.com/doggy-dentures/Funkin
> 
> good luck

Can I set it with kade engine?

**Reply by Riskerman21** [reply]
not that code sorrry

**Comment by Reyororo** [comment, marked as answer]
Thanks to everyone, I succeeded
I appreciate you sincerely
This best answer belongs to everyone who cooperated

https://user-images.githubusercontent.com/88378158/128627111-10f6d1c3-42b3-4159-b542-3ed0690bcca9.mp4
